services/libs/aster-std/src/vm/vmo/mod.rs

```diff
@@ -127,198 +133,305 @@ bitflags! {
+    /// 3. For a File-backed Vmo that requires triggering the COW mechanism, obtain a page from the pager and then copy it. This page can be set as Exclusive.
+    fn prepare_page(
+        &self,
+        page_idx: usize,
+        need_cow: bool,
+        will_write: bool,
+    ) -> Result<(VmFrame, bool)> {
+        let (page, is_exclusive) = match &self.pager {
+            None => {
+                // Condition 1.
+                (VmAllocOptions::new(1).alloc_single()?, need_cow)
+            }
+            Some(pager) => {
+                let page = pager.commit_page(page_idx)?;
+                // The prerequisite for triggering the COW mechanism here is that the current Vmo requires COW and the prepared page is about to undergo a write operation.
+                // At this point, the VmFrame obtained from the pager needs to be cloned to avoid subsequent modifications affecting the content of the VmFrame in the pager.
+                let trigger_cow = need_cow && will_write;
+                // Condition 3.
+                if trigger_cow {
+                    (clone_page(&page)?, true)
+                // Condition 2.
+                } else {
+                    (page, false)
+                }
+            }
         };
-        self.insert_frame(page_idx, frame);
-        Ok(())
+        Ok((page, is_exclusive))
     }
 
-    fn decommit_page(&mut self, offset: usize) -> Result<()> {
-        let page_idx = offset / PAGE_SIZE;
-        if self.committed_pages.remove(&page_idx).is_some() {
-            if let Some(pager) = &self.pager {
-                pager.decommit_page(page_idx)?;
+    /// Commit the page corresponding to the target offset in the Vmo and return that page.
+    /// If the current offset has already been committed, the page will be returned directly.
+    /// During the commit process, the Copy-On-Write (COW) mechanism may be triggered depending on the circumstances.
+    pub fn commit_page(&self, offset: usize, will_write: bool) -> Result<VmFrame> {
+        let page_idx = offset / PAGE_SIZE + self.page_idx_offset;
+        let mut pages = self.pages.lock();
+        let need_cow = pages.is_marked(VmoMark::NeedCow);
+        let mut cursor = pages.cursor_mut(page_idx as u64);
+        let (new_page, is_exclusive) = {
+            if let Some(committed_page) = cursor.load() {
+                let is_exclusive = cursor.is_marked(VmoMark::Exclusive).unwrap();
+                // The necessary and sufficient condition for triggering the COW mechanism is that
+                // the current Vmo requires copy-on-write, there is an impending write operation to the page, and the page is not exclusive.
+                let trigger_cow = need_cow && will_write && !is_exclusive;
+                if !trigger_cow {
+                    // Fast path: return the page directly.
+                    return Ok(committed_page.clone());
+                }
+
+                (clone_page(committed_page)?, true)
```

Comment:
Why is the statement `(clone_page(committed_page)?, true)` instead of `(clone_page(committed_page)?, is_exclusive)`?